src/participant_network.star

```diff
@@ -67,9 +70,115 @@ def launch_participant_network(
     global_tolerations,
     parallel_keystore_generation=False,
 ):
+    network_id = network_params.network_id
     num_participants = len(participants)
-    if network_params.network == constants.NETWORK_NAME.kurtosis:
-        # We are running a kurtosis network
+    latest_block = ""
+    cancun_time = 0
+    prague_time = 0
+    if (
+        network_params.network == constants.NETWORK_NAME.kurtosis
+        or constants.NETWORK_NAME.shadowfork in network_params.network
+    ):
+        if (
+            constants.NETWORK_NAME.shadowfork in network_params.network
+        ):  # shadowfork requires some preparation
```

Comment:
this method is continously getting bigger; and I would like to get this refactored. It's at 640~ lines; lets refactor it into smaller parts that make sense

Reply:
I'd focus on getting this merged, and refactor in an upcoming PR.

Reply:
tracking it here: https://github.com/kurtosis-tech/ethereum-package/issues/492